src/blueapi/cli/cli.py

```diff
@@ -50,14 +59,25 @@ def main(ctx: click.Context, config: Path | None | tuple[Path, ...]) -> None:
             else:
                 raise FileNotFoundError(f"Cannot find file: {path}")
 
+    # Step 3: Load values from environment variables
+    config_loader.use_values_from_env(env_prefix)
+
+    # Step 4: Load CLI arguments as overrides
+    config_loader.use_values_from_cli(ctx.params)
+
+    # Load the final configuration
     ctx.ensure_object(dict)
     loaded_config: ApplicationConfig = config_loader.load()
 
+    # Store config in the Click context object for subcommands to access
     ctx.obj["config"] = loaded_config
+
+    # Set up logging based on loaded configuration
     logging.basicConfig(level=loaded_config.logging.level)
 
+    # If no subcommand is invoked, print a help message
```

Comment:
Nit: The help message is taken from `--help` and is not the same as this

```suggestion
    # If no subcommand is invoked, print an error message
```